Test.c

```diff
@@ -4,11 +4,151 @@
 // Ayisha Sameera,GET.
 // ------------------------------------------------------------------
 // Test.c
-// Program on branch main
+// Program on branch A5
 // ------------------------------------------------------------------------------------------------
-
+#pragma warning(disable:4996)
+#include "Header.h"
 #include <stdio.h>
+#include <stdbool.h>
+#include <malloc.h>
+#include <stdlib.h>
+#include <limits.h>
+
+#define MAX_SIZE 20
+
+typedef struct {
+   int Array[MAX_SIZE];
+   int Length;
+}Test;
+
+/// <summary>Checks if the array is sorted in ascending order.</summary>
+bool IsSorted (int arr[], int length) {
+   for (int i = 1; i < length; i++) if (arr[i - 1] > arr[i]) return false;
+   return true;
+}
+
+void PrintArray (int arr[], int size, int expLength) {
+   for (int i = 0; i < size; i++) printf ("%5d", arr[i]);
+   for (int i = expLength - size; i > 0; i--) printf ("%5s", "");
+}
+
+void PrintLine () {
+   printf ("--------------------------------------------------------------------------");
+   printf ("------------------------------------------------------------------------\n");
+}
+
+void TestSortAndSearch () {
+   Test arr[] = { { { 1,2,3,5,4,6,7,8 } , 8 },
+                  { { 8,7,5,4,3,1 } , 6 } ,
+                  { { -1,-3,-5,-9,-234,-90,-123,0 } , 8 },
+                  { { 0,5,-567,67,8,2,9,124 } , 8 } ,
+                  { { 1,1,1,1,1,1,1 } , 7} ,
+                  { { 2 } ,1 } };
+   int search[][2] = { { 3 , 2 } , { 8 , 5 },
+                       { -9 , 3 } , { 9 , 5 },
+                       { 1 , 0 } , { 5 , -1 } };
+   int length = sizeof (arr) / sizeof (arr[0]);
+   PrintLine ();
+   printf ("|                  Input                  |                   Output        ");
+   printf ("          |  Sort  ||   Key  | Found at |     Status      |  Search  | \n");
+   for (int i = 0; i < length; i++) {
+      PrintLine ();
+      printf ("|");
+      PrintArray (arr[i].Array, arr[i].Length, 8);
+      HeapSort (arr[i].Array, arr[i].Length);
+      printf (" | ");
+      PrintArray (arr[i].Array, arr[i].Length, 8);
+      printf ("  |  %6s  ",
+              IsSorted (arr[i].Array, arr[i].Length) ? CYAN"Pass"RESET : MAGENTA"Fail"RESET);
+      int index = BinarySearch (arr[i].Array, arr[i].Length, search[i][0]);
+      printf ("||  %4d  |  %4d    ", search[i][0], index);
+      printf ("|    %10s    ", index == -1 ? BLUE"Not found"RESET : CYAN"  Found  "RESET);
+      printf ("|   %6s   |\n", search[i][1] == index ? CYAN"Pass"RESET : MAGENTA"Fail"RESET);
+   }
+   PrintLine ();
+}
+
+/// <summary>Gets user input and returns true if 'y' is pressed.</summary>
+bool Choice () {

```

Comment:
![image](https://github.com/user-attachments/assets/27137d36-a364-4c4c-a0d3-b438d448202d)
Please try to incorporate changes suggested by intellisense wherever appropriate